### PR TITLE
Refactor CLI so that additional commands won't need common code updates

### PR DESCRIPTION
**Pull request by mdeuser**
Desire is for new commands to be completely implemented in a couple command-specific files without any impact to common code (i.e. utilities, CLI framework).

At this point, the globalized content is still in a single file..

### CODE REVIEW

**Comment by mdeuser** [comment]
recycle PR state to re-trigger failed travis build (travis "restart build" did not work)

**Comment by mdeuser** [comment]
PG3 290 - encountered cloundant slowness that failed an API related test
PG3 292

File: tools/cli/go-whisk/whisk/client.go [hunk 1]
```diff
@@ -435,60 +438,100 @@ func IsResponseResultSuccess(data []byte) bool {
...
+        Debug(DbgError, "http.NewRequest(%v, %s, buf) error: %s\n", method, requestUrl.String(), err)
         errStr := wski18n.T("Error initializing request: {{.err}}", map[string]interface{}{"err": err})
         werr := MakeWskError(errors.New(errStr), EXITCODE_ERR_GENERAL, DISPLAY_MSG, NO_DISPLAY_USAGE)
         return nil, werr
     }
-    if req.Body != nil {
+    if (req.Body != nil && encodeBodyAs == EncodeBodyAsJson) {
         req.Header.Add("Content-Type", "application/json")
     }
+    if (req.Body != nil && encodeBodyAs == EncodeBodyAsFormData) {
```
Comment (dubeejw) [inline tools/cli/go-whisk/whisk/client.go]: Might be more efficient to check if the request body is nil only once here.

File: tools/cli/go-whisk/whisk/info.go [hunk 2]
```diff
@@ -36,18 +37,16 @@ type InfoService struct {
...
-
-    ref, err := url.Parse(s.client.Config.Version)
+    urlStr := fmt.Sprintf("%s/%s", s.client.BaseURL.String(), s.client.Config.Version)
+    u, err := url.Parse(urlStr)
     if err != nil {
-        Debug(DbgError, "url.Parse(%s) error: %s\n", s.client.Config.Version, err)
+        Debug(DbgError, "url.Parse(%s) error: %s\n", urlStr, err)
         errStr := wski18n.T("Unable to URL parse '{{.version}}': {{.err}}",
-            map[string]interface{}{"version": s.client.Config.Version, "err": err})
+            map[string]interface{}{"version": urlStr, "err": err})
```
Comment (dubeejw) [inline tools/cli/go-whisk/whisk/info.go]: Looks like version isn't the correct parameter name here anymore.

**Comment by dubeejw** [comment]
Besides the two comments above, the PR looks good since we already reviewed it.

**Comment by mdeuser** [comment]
@csantanapr - review complete.  travi/pg passed.  ready for merging, please

File: tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json [hunk 3]
```diff
@@ -1354,5 +1354,77 @@
   {
     "id": "display full description of each API",
     "translation": "display full description of each API"
+  },
+  {
+    "id": "User name and password were not specified",
+    "translation": "User name and password were not specified"
+  },
+  {
+    "id": "Unable to access Bluemix API host: {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json]: why is openwhisk talking about a bluemix api host?

**Comment by rabbah** [comment]
why are there references to Bluemix here?

**Comment by csantanapr** [comment]
@mdeuser see comment
> why are there references to Bluemix here?

**Comment by mdeuser** [comment]
Looks like the globalized strings must be merged during the build process as well..

**Comment by mdeuser** [comment]
PG3 324
@csantanapr - good to merge

**Comment by csantanapr** [comment]
merged 2861cf02cb8c6cd5cf60a94db1eb00812c283515

**Pull request closed without merging**